### DISCUSSION TITLE
API Token Expires on Container Restart

### DISCUSSION THREAD

**Discussion by ackstorm23** [TechnitiumSoftware/DnsServer#441] [Q&A] [answered]
My experience creating API tokens has been that if the docker container is restarted, the tokens expire.   

This makes it very painful to try and monitor the service through the API, as any restart of the container requires re-acquiring an API token. 

Is this by design, or am I experiencing something that should not be?

**Comment by ShreyasZare** [comment, marked as answer]
Thanks for asking. This is how the current implementation is. The next major update is adding multi-user support with fixed tokens to use with API automation stuff. So this will get solved soon.
